### ISSUE TITLE
`no-synchronous-tests` rule reports false positive on synchronous `describe` callbacks

### ISSUE THREAD

**Issue by jsn-xciv** [lo1tuma/eslint-plugin-mocha#399]
# Description

The `no-synchronous-tests` rule incorrectly flags the callback function in a top-level `describe` block as a "synchronous test", even though `describe` suites must always be synchronous (Mocha executes them immediately for registration). The rule should only target test cases like `it` or `specify`.

This is a false positive and produces lint noise in test files with nested suites.

# Reproduction

1. Create a new Node.js project with ESLint 9.x flat config.

2. Install dependencies: `npm install --save-dev eslint@^9.34.0 eslint-plugin-mocha@^11.1.0 globals@^16.3.0 @eslint/js@^9.34.0 mocha@^11.7.2`.

3. Add the provided `eslint.config.mjs` (below).

4. Create a test file `test/sample.spec.js` with the provided code.

5. Run `npx eslint test/sample.spec.js`, reports the error on line 1:24 in the `describe` callback.

# Expected behavior

No error, as `describe` callbacks are inherently synchronous and not tests.

# Actual behavior

```
test/sample.spec.js
  1:24  error  Unexpected synchronous test  mocha/no-synchronous-tests
```

# Environment

* ESLint: 9.34.0
* eslint-plugin-mocha: 11.1.0
* Node.js: v24.7.0
* OS: Linux

# Config

```js
import { defineConfig } from 'eslint/config';
import globals from 'globals';
import js from '@eslint/js';
import mochaPlugin from 'eslint-plugin-mocha';

export default defineConfig([
	mochaPlugin.configs.recommended,
	{
		files: ['**/*.{js,mjs,cjs}'],
		...js.configs.recommended,
		languageOptions: {
			globals: { ...globals.node }
		},
	},
	{
		files: ['test/**/*.js'],
		languageOptions: {
			globals: { ...globals.mocha, mocha: 'readonly' },
			sourceType: 'commonjs',
		},
		rules: {
			'mocha/no-synchronous-tests': 'error',
		},
	},
]);
```

# Code example

`test/sample.spec.js`:

```js
describe('Some tests', function () {
	it('should do something', async function () {
		await Promise.resolve(true);
	});

	it('should do something else', async function () {
		await Promise.resolve(true);
	});
});
```

Thanks for looking into this!
